### ISSUE TITLE
実験: Frozen競り相手をConservative / All-Pass 50:50混合に変更してPASS collapseを再検証

### ISSUE THREAD

**Issue by eletim** [eletim/napoleon-web#360]
Related: #202, #358, #359
Base branch: `feature/issue-202-non-playing-ai`

## 背景

Issue #358 / #359 の調査で、最新 bidding PPO は以下の状態であることが分かった。

- deterministic argmax では opponent roster に関係なく PASS top-1 = 100%
- stochastic temperature=1 では 25〜28% 程度 bid するが、これは PASS top-1 分布からのsampling
- current RuleBased / Conservative 50:50 mix では、candidateがPASSした後に他プレイヤーがNapoleonを引き受け、candidateがCitizen/Adjutantとして比較的安全なrewardを得る構造がある
- Conservative比率はAll Pass率には強く影響するが、candidate自身のPASS優位の主因ではない
- Reward v3 relative + current opponent mixでは PASS sample の advantage が bid sample より大幅に高い

一方、現在の `RuleBased / Conservative 50:50` は両者とも競りに参加する可能性があり、candidateから見ると「誰かがNapoleonを引き受けてくれる」環境がまだ比較的多い。

そこで、Frozen bidding opponentを明確に二極化し、

- 強い手なら競りに参加する `ConservativeBiddingAgent`
- biddingでは常にPASSする `AllPassBiddingAgent`

の50:50混合へ変更して、candidateが「自分が競らないと誰も契約を取らない可能性がある」環境を十分経験するようにする。

## 目的

Frozen bidding opponents 4席を、各seat independently / seeded deterministic に

- ConservativeBiddingAgent: 50%
- AllPassBiddingAgent: 50%

で選択する構成へ変更し、reward / PPO / network / observation / playing policyを変えずに bidding PPO を再学習して、PASS collapseが改善するか検証する。

## All-Pass baseline

新しい学習modelは作らない。

`AllPassBiddingAgent` は bidding phase では合法な限り常に PASS を選択する固定方策とする。

要件:

- biddingでは常にPASS
- deterministic
- 学習中に更新されないFrozen policy
- bidding以外のphaseへ不要な影響を与えない
- 全員PASS時などgame-core側の既存ルール処理はそのまま利用する

既存に同等のagent / policyが存在する場合は再利用し、重複実装しない。

## 50:50混合方法

Frozen 4席について、各seatごとに independently 50:50 で

- Conservative
- All-Pass

を選ぶ。

ゲーム全体で4席すべてを同一baselineへ固定する方式にはしない。

乱数選択は既存opponent mixと同様に seed から決定論的に再現可能であること。

4席すべてがAll-Passになるケースも自然に発生すること。

## 維持する条件

このIssueでは opponent mix 以外を変更しない。

- candidate x1 + Frozen x4 topology
- candidate seat 5席rotation
- Reward v3
- relative reward transform `raw_reward_i - mean(raw_reward_all_players)`
- all-pass payoff / terminal handling
- PPO algorithm
- PPO hyperparameters
- bidding observation schema / compact278 model
- Frozen playing policy `ppo-separated-v1000`
- adjutant / exchange artifacts
- temperature=1 rollout sampling
- strongest-suit-balanced minibatch strategy

特に、今回の実験ではAll Pass rewardを変更しない。まず opponent distribution だけの効果を切り分ける。

## provenance / resume safety

manifest / run config / provenance から新しいopponent mixを明確に識別できるようにする。

最低限:

- Conservative policy identifier
- All-Pass policy identifier
- mix rule/version
- seat単位のseeded deterministic selection
- Conservative seat count / All-Pass seat count

既存の RuleBased / Conservative 50:50 run を新mixで誤resumeしないよう fail-close する。

## 事前確認

学習前に固定seedで少なくとも以下を確認する。

1. Conservative 100%
2. All-Pass 100%
3. Conservative / All-Pass 50:50

確認項目:

- opponent bidding PASS率
- mix時のseat countが概ね50:50であること
- deterministic再現性
- fallback=0 / illegal action=0

## 再学習

まず短期runで挙動を確認する。

推奨:

```text
iterations             30
games-per-iteration    200 logical seeds
epochs                  current setting
batch-size              current setting
learning-rate           current setting
evaluation interval     current setting
seed                    202
```

ただし current compact278 bidding-only run と比較できるよう、可能な限り現在の実行条件をそのまま使う。

30 iterationsで方向が良好なら、その後100 iterations相当まで延長するかは結果を見て判断する。

旧runは上書きしない。

新しいoutput directoryは既存runと混同しない名称にする。

## 必須diagnostics

Issue #358 / #359 で問題が判明したため、最低限以下を各iterationまたはevaluation checkpointで追跡する。

- PASS rate
- bid rate
- candidate Napoleon formation rate
- declaration success rate
- all-pass rate
- candidate role distribution
- mean raw / transformed reward
- PASS top-1 rate
- argmax bid rate
- PASS probability
- best legal bid probability
- PASS-bestBid logit gap
- policy entropy

可能なら:

- PASS vs bid advantage
- role別return / advantage
- hand-strength bin別PASS / bid挙動
- strongest suit match rate

## 比較対象

主比較対象は current compact278 run:

`/home/eletim/napoleon_runs/non-playing-ppo-compact278-bidding-only-stratified-suit32-e01-lr1e4-100x2000`

同じiteration付近で最低限以下を比較する。

- stochastic PASS rate
- PASS top-1 rate
- argmax bid rate
- PASS-bestBid gap
- candidate Napoleon rate
- All Pass rate
- declaration success rate

## 成功判定

単純にPASS率が低ければ成功とはしない。

望ましい方向:

- argmax bid が0%から明確に増える
- PASS top-1が100%張り付きから外れる
- 強いhandでbid top-1が発生する
- candidate Napoleon rateが0% / 極端な低値から回復する
- stochastic PASS率が極端な0%または100%へ崩壊しない
- All Passが増えても、それを避けるためcandidateが無条件bidへ崩壊しない
- strongest suitへの追従性が少なくとも悪化しない

## 判定したいこと

最終報告では以下へ明示的に回答する。

1. Conservative / All-Pass 50:50 にすると PASS collapse は改善するか
2. PASS top-1 100% は解消するか
3. argmaxで実際にbidするようになるか
4. 強いhandほどbid top-1になりやすくなるか
5. All Pass頻度はどの程度増えるか
6. candidate Napoleon率 / declaration successはどう変化するか
7. rewardを変えず opponent distributionだけで十分改善するか
8. 改善しない場合、次はAll Pass reward / bidding credit assignmentを変更すべきか

## Git / PR運用

このIssueは opponent mix の実装変更を含むため、Issue専用ブランチを最新 `feature/issue-202-non-playing-ai` から作成する。

PR baseは `feature/issue-202-non-playing-ai` とする。

実装・smoke・必要な短期実験完了後、Codex review最大4回（review後修正最大3回）を行う。

blocking / actionable 指摘がなく、必要テストが成功したら Ready for review にして親ブランチへmergeする。

`main`にはmergeしない。

実験結果はIssueへコメントし、必要なら `experiments/` 配下へ記録する。

## Acceptance Criteria

- Frozen bidding opponent が Conservative / All-Pass 50:50 になる
- 各Frozen seat independently / seeded deterministic に選択される
- All-Pass baseline は biddingで常にPASSする
- candidate x1 + Frozen x4 + 5-seat rotationを維持
- Reward / PPO / network / observation / playing policyを変更しない
- opponent mix provenanceを追跡可能
- 旧runを誤resumeしない
- fixed-seed baseline確認が通る
- smokeで fallback=0 / illegal action=0
- 少なくとも短期再学習を実施し、current compact278 runと比較する
- PASS top-1 / argmax bid / PASS-bestBid gapを評価する
- 次の方針を判断できる具体的な実験結果をIssueに残す

## 非目標

- Reward係数変更
- All Pass reward変更
- bidding credit assignment変更
- PPO hyperparameter変更
- network architecture変更
- observation schema変更
- playing policy再学習
- mainへのMerge
